app/models/user.rb

```diff
@@ -391,6 +396,10 @@ def email_uniqueness
     email_uniqueness_errors(existing_user)
   end
 
+  def check_email_and_name
+    errors.add(:name, "can't be the same as email") if email == name
```

Comment:
This isn't strong enough a check; I saw a case where the email and name only differed by capitalisation.  We need to prevent `@` in the name.